jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -285,6 +289,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
+    private List<String> getPreferredEncodingOrder(HttpServletRequest request)
+    {
+        Enumeration<String> e = request.getHeaders(HttpHeader.ACCEPT_ENCODING.asString());
+        if (e==null || !e.hasMoreElements())
+            return emptyList();
+
+        String key=e.nextElement();
+        while (e.hasMoreElements())
+            key+='\n'+e.nextElement();
+
+        List<String> values=_preferredEncodingOrderCache.get(key);
+        if (values==null)
+        {
+            QuotedEncodingQualityCSV encodingQualityCSV = new QuotedEncodingQualityCSV(_precompressedFormats);
```

Comment:
The encoding->index Trie would work. But because the array length of known encodings is 2 (in the default config, and I'm sure no-one will configure more than 4) then is iterating that array too slow? There is no object creations inside the secondary sorting function.